flink/src/main/java/org/apache/iceberg/flink/data/vectorized/VectorizedFlinkOrcReaders.java

```diff
@@ -0,0 +1,381 @@
+              @Override
+              public double getDouble(int i) {
+                return (double) constant;
+              }
+
+              @Override
+              public boolean isNullAt(int i) {
+                return false;
+              }
+            };
+
+          case FLOAT:
+            return new org.apache.flink.table.data.vector.FloatColumnVector() {
+              @Override
+              public float getFloat(int i) {
+                return (float) constant;
+              }
+
+              @Override
+              public boolean isNullAt(int i) {
+                return false;
+              }
+            };
+
+          case DECIMAL:
+            return new org.apache.flink.table.data.vector.DecimalColumnVector() {
+              @Override
+              public DecimalData getDecimal(int i, int precision, int scale) {
+                return (DecimalData) constant;
+              }
+
+              @Override
+              public boolean isNullAt(int i) {
+                return false;
+              }
+            };
+
+          case DATE:
+            return new org.apache.flink.table.data.vector.IntColumnVector() {
+              @Override
+              public int getInt(int i) {
+                return (int) constant;
+              }
+
+              @Override
+              public boolean isNullAt(int i) {
+                return false;
+              }
+            };
+
+          case TIME:
+            return new org.apache.flink.table.data.vector.IntColumnVector() {
+              @Override
+              public int getInt(int i) {
+                return (int) constant;
+              }
+
+              @Override
+              public boolean isNullAt(int i) {
+                return false;
+              }
+            };
+
+          case TIMESTAMP:
+            return new org.apache.flink.table.data.vector.TimestampColumnVector() {
+              @Override
+              public TimestampData getTimestamp(int i, int precision) {
+                return (TimestampData) constant;
+              }
+
+              @Override
+              public boolean isNullAt(int i) {
+                return false;
+              }
+            };
+
+          case STRING:
+            return new org.apache.flink.table.data.vector.BytesColumnVector() {
+              @Override
+              public Bytes getBytes(int i) {
+                BinaryStringData str = (BinaryStringData) constant;
+                return new Bytes(str.toBytes(), 0, str.getSizeInBytes());
+              }
+
+              @Override
+              public boolean isNullAt(int i) {
+                return false;
+              }
+            };
+
+
+          default:
+            throw new UnsupportedOperationException("Unsupported data type for constant.");
+        }
+      } else {
+        throw new UnsupportedOperationException("ConstantColumnVector only supports primitives.");
+      }
+    }
```

Comment:
The [comment](https://github.com/apache/iceberg/pull/2566/files#r629023952) should be also applied to this section.